pilot/pkg/networking/core/httproute.go

```diff
@@ -257,7 +257,7 @@ func selectVirtualServices(virtualServices []config.Config, servicesByName map[h
 			// TODO: This is a bug. VirtualServices can have many hosts
 			// while the user might be importing only a single host
 			// We need to generate a new VirtualService with just the matched host
-			if servicesByName[host.Name(h)] != nil {
+			if servicesByName[host.Name(strings.ToLower(string(h)))] != nil {
```

Comment:
This is the actual fix for this issue. Virtual service is not selected because of this. But added the same at other places for consitency

Reply:
silly question: doesn't envoy already handle hosts having mixed casing?

Reply:
The problem is because of this host mismatch - we do not pick the virtual service and apply default routes. When we send domains to Envoy, we send it lower case.